apiplugin/file/file_test.go

```diff
@@ -1,26 +1,193 @@
 package main
 
-import "testing"
+import (
+	"context"
+	"log"
+	"testing"
+
+	pcontext "github.com/iansmith/parigot/context"
+	"github.com/iansmith/parigot/g/file/v1"
+)
 
 // if you look at the tests in queue_test you'll see that
 // the tests ignore the wrapper functions and use the
 // real impleentations directly.
 
+const fileContent = "Hello! Parigot!"
+const filePath = "/parigot/app/file.go"
+
 func TestOpenClose(t *testing.T) {
+	svc := newFileSvc((context.Background()))
+	creatAGoodFile(t, svc)
 
-	// try open a file with a badly formed path name, make sure it fails
-	// see filepath.Clean()
+	// try opening a file does not exist
+	badFid := testFileOpen(t, svc, "/parigot/app/badfile.txt", "path doesn't exist",
+		true, int32(file.FileErr_NotExistError))
```

Comment:
This is a good use of the error codes!  It's great when you can test for the exact error, not just "some" error.